flake: TestTxClientTestSuite/TestGasConsumption flakes on balance query after SubmitTx

## Summary

Follow-up to [PROTOCO-1519](https://linear.app/celestia/issue/PROTOCO-1519/flake-testtxclienttestsuite-fails-with-tx-not-found-after-submittx). `TestTxClientTestSuite/TestGasConsumption` still flakes on `main` after PR celestiaorg/celestia-app#7095, because PR celestiaorg/celestia-app#7095 only fixed the tx-indexer race for `ServiceClient.GetTx`. The remaining race is in the bank gRPC balance query `queryCurrentBalance` that is called immediately after `SubmitTx`.

## Failing run

* Workflow: Post-merge CI
* Run: [https://github.com/celestiaorg/celestia-app/actions/runs/24572644548/job/71849614942](<https://github.com/celestiaorg/celestia-app/actions/runs/24572644548/job/71849614942>)
* Commit: `0d20747f3172601d605474369ad805591baa81cc` (docs: add v7.x and v8.x to dependency forks table)
* Job: `test / go-test (github.com/celestiaorg/celestia-app/v9/pkg/user)`

## Failure

```
--- FAIL: TestTxClientTestSuite (40.20s)
    --- FAIL: TestTxClientTestSuite/TestGasConsumption (2.01s)
        tx_client_test.go:610:
            Error Trace: pkg/user/tx_client_test.go:610
            Error:       Not equal
                         expected: 1000000
                         actual  : -1
            Test:        TestTxClientTestSuite/TestGasConsumption
```

With `utiaToSend = 1`, `amountDeducted = -1` means `balanceBefore == balanceAfter`; the post-`SubmitTx` balance query returned stale (pre-tx) state.

## Root cause

`TxClient.ConfirmTx` returns as soon as CometBFT's `TxStatus` RPC reports the tx as committed. At that point CometBFT's blockstore has advanced, but the app's multistore `Commit()` that makes the new state visible over gRPC may still be in flight (see the comment in `test/util/testnode/node_interaction_api.go` `WaitForHeightWithTimeout`: *"CometBFT's blockStore.Height() advances before the app's cms.Commit() runs, so gRPC queries can fail … if we return too early"*).

`queryCurrentBalance` calls `bank.QueryClient.AllBalances` directly, so when it races the post-commit window it returns the pre-tx balance and the deduction assertion fails.

PR celestiaorg/celestia-app#7095's `GetTxWithRetry` only helps for calls to `ServiceClient.GetTx` (which return `NotFound` until the indexer catches up). The balance query returns *stale data*, not `NotFound`, so it is not covered.

## Proposed fix

After `SubmitTx` returns in `TestGasConsumption`, wait for the app's multistore to reach `resp.Height` before querying the post-tx balance. `testnode.Context.WaitForHeight` already checks `ABCIInfo().LastBlockHeight`, so using it guarantees the bank gRPC query sees the tx's block.

## Related

* [PROTOCO-1519](https://linear.app/celestia/issue/PROTOCO-1519/flake-testtxclienttestsuite-fails-with-tx-not-found-after-submittx) (closed by celestiaorg/celestia-app#7095)

--- a/pkg/user/tx_client_test.go
+++ b/pkg/user/tx_client_test.go
@@ -603,6 +603,14 @@ func (suite *TxClientTestSuite) TestGasConsumption() {
 	require.NoError(t, err)
 
 	require.EqualValues(t, abci.CodeTypeOK, resp.Code)
+
+	// SubmitTx returns as soon as CometBFT reports the tx committed, but the
+	// app's multistore Commit() that makes the post-tx state visible over gRPC
+	// may still be in flight. Wait for the app to reach resp.Height so the
+	// balance query below reflects the tx.
+	_, err = suite.ctx.WaitForHeight(resp.Height)
+	require.NoError(t, err)
+
 	balanceAfter := suite.queryCurrentBalance(t)
 
 	// verify that the amount deducted depends on the fee set in the tx.

